crates/red_knot_python_semantic/src/types.rs

```diff
@@ -482,26 +495,39 @@ impl<'db> Type<'db> {
     ///
     /// Returns `None` if `self` is not a callable type.
     #[must_use]
-    pub fn call(&self, db: &'db dyn Db) -> Option<Type<'db>> {
+    fn call(self, db: &'db dyn Db, arg_types: &[Type<'db>]) -> CallOutcome<'db> {
         match self {
-            Type::Function(function_type) => Some(function_type.return_type(db)),
+            // TODO validate typed call arguments vs callable signature
+            Type::Function(function_type) => CallOutcome::callable(function_type.return_type(db)),
+            Type::RevealTypeFunction(function_type) => CallOutcome::RevealType {
+                return_ty: function_type.return_type(db),
```

Comment:
Is this a fixed return type or is the `reveal_type` function similar to `dbg` where it is an `identity` function with the side effect that it prints the type?

Reply:
It's a generic identity function, the same as `dbg!`: https://github.com/python/cpython/blob/32119fc377a4d9df524a7bac02b6922a990361dd/Lib/typing.py#L3635-L3651

Reply:
It would be possible here to instead set `return_ty` to the same as the revealed type. But this is another case where I would prefer to have less special-casing, and allow the return type to be inferred via the typeshed stubs, the same as for any other function.